allennlp/common/file_utils.py

```diff
@@ -233,9 +235,45 @@ def cached_path(
     cache_dir = os.path.expanduser(cache_dir)
     os.makedirs(cache_dir, exist_ok=True)
 
+    extraction_path: Optional[str] = None
+
     if not isinstance(url_or_filename, str):
         url_or_filename = str(url_or_filename)
 
+    if not os.path.isfile(url_or_filename) and not urlparse(url_or_filename).scheme in ("http", "https", "s3"):
+        # If can't be resolved to a path or a URL,
+        # let's try to find it on Hugging Face model hub
+        # e.g. lysandre/pair-classification-roberta-mnli is a valid model id
+        # and  lysandre/pair-classification-roberta-mnli@main supports specifying a commit/branch/tag.
+        if len(url_or_filename.split("/")) > 2:
+            filename = url_or_filename.split("/")[2]
+            url_or_filename = "/".join(url_or_filename.split("/")[:2])
+        else:
+            filename = None
+
+        if "@" in url_or_filename:
+            repo_id = url_or_filename.split("@")[0]
+            revision = url_or_filename.split("@")[1]
+        else:
+            repo_id = url_or_filename
+            revision = None
+
+        try:
+            if filename is not None:
+                url = hf_hub_url(
+                    repo_id=repo_id, filename=filename, revision=revision
+                )
+                url_or_filename = cached_download(
+                    url=url,
+                    library_name="allennlp",
+                    library_version=VERSION,
+                    cache_dir=CACHE_DIRECTORY,
+                )
+            else:
+                extraction_path = Path(snapshot_download(repo_id, revision=revision, cache_dir=CACHE_DIRECTORY))
+        except HTTPError as e:
+            logger.warning(f"Tried to download from Hugging Face Hub but failed with {e}")
```

Comment:
Why warn here and not just let the exception propagate?

I think it would be better to emit a `logger.info()` statement above that says "treating url_or_filename as a HuggingFace Hub resource" or something, and then just let this error propagate if it happens.